packages/babel-plugin/src/keyframes/__tests__/call-expression.test.tsx

```diff
@@ -0,0 +1,1157 @@
+
+            const fadeOut = keyframes({
+              from: {
+                color: from().color,
+                opacity: from().opacity,
+              },
+              to: {
+                color: to().color,
+                opacity: to().opacity,
+              },
+            });
+
+            <div css={{ animationName: fadeOut }} />
+          `);
+
+          expect(actual).toMatchInlineSnapshot(`
+            "const _2 = \\"._j7hqwahf{animation-name:k1fh39u1}\\";
+            const _ =
+              \\"@keyframes k1fh39u1{0%{color:blue;opacity:1}to{color:indigo;opacity:0}}\\";
+
+            const from = () => ({
+              color: \\"blue\\",
+              opacity: 1,
+            });
+
+            const to = () => ({
+              color: \\"indigo\\",
+              opacity: 0,
+            });
+
+            const fadeOut = null;
+            <CC>
+              <CS>{[_, _2]}</CS>
+              {<div className={ax([\\"_j7hqwahf\\"])} />}
+            </CC>;
+            "
+          `);
+        });
+
+        it('through an identifier that references an arrow function call expression', () => {
+          const actual = transform(`
+            import { keyframes } from '@compiled/react';
+
+            const getFrom = () => ({ color: 'blue', opacity: 1 });
+            const getTo = () => ({ color: 'indigo', opacity: 0 });
+
+            const from = getFrom();
+            const to = getTo();
+
+            const fadeOut = keyframes({
+              from: {
+                color: from.color,
+                opacity: from.opacity,
+              },
+              to: {
+                color: to.color,
+                opacity: to.opacity,
+              },
+            });
+
+            <div css={{ animationName: fadeOut }} />
+          `);
+
+          expect(actual).toMatchInlineSnapshot(`
+            "const _2 = \\"._j7hqgafb{animation-name:k1qk0qiq}\\";
+            const _ =
+              \\"@keyframes k1qk0qiq{0%{color:blue;opacity:1}to{color:indigo;opacity:0}}\\";
+
+            const getFrom = () => ({
+              color: \\"blue\\",
+              opacity: 1,
+            });
+
+            const getTo = () => ({
+              color: \\"indigo\\",
+              opacity: 0,
+            });
+
+            const from = getFrom();
+            const to = getTo();
+            const fadeOut = null;
+            <CC>
+              <CS>{[_, _2]}</CS>
+              {<div className={ax([\\"_j7hqgafb\\"])} />}
+            </CC>;
+            "
+          `);
+        });
+      });
+
+      describe('by inlining suffixes prefixed by an', () => {
+        it('identifier that references a number literal', () => {
+          const actual = transform(`
+            import { keyframes } from '@compiled/react';
+
+            const fromFontSize = 14;
+            const toFontSize = 18;
+
+            const enlargeFont = keyframes({
+              from: {
+                fontSize: \`\${fromFontSize}px\`,
```

Comment:
I haven't tried yet, but it seems that `addUnitIfNeeded` should deal with missing units. Maybe we could omit `px ` from one of these declarations and see what happens :)

Reply:
Yes, it does deal with this case. Omitting it just use the value `fromFontSize`. I won't add tests for these under keyframes, since the `addUnitIfNeeded` logic should be tested more thoroughly elsewhere.